scripts/checkDeps.ps1

```diff
@@ -40,48 +38,129 @@ if ($? -eq $false || [string]::IsNullOrEmpty($_dockerComposeV)) {
     exit 69
 }
 
+# Get the deps at .conf/deps.json file
+$_deps = Get-Content .conf/deps.json | ConvertFrom-Json
+
+
 # ok docker and docker-compose exist so let's check the packages
 $_packagesToInstall = New-Object Collections.Generic.List[string]
 
+
+# Check if all dependencies are installed
 Write-Host -ForegroundColor Yellow "Checking dependencies ..."
 
-foreach ($package in $_packages.packages) {
+foreach ($package in $_deps.packages) {
+
     dpkg -s $package > /dev/null 2>&1
-    
+
     if ($? -eq $false) {
         $_packagesToInstall.Add($package)
-        Write-Host -ForegroundColor DarkRed "😵 $package not installed"
+        Write-Host -ForegroundColor DarkRed "😵 $package debian package dependency not installed"
+    } else {
+        Write-Host -ForegroundColor DarkGreen "👍 $package debian package dependency installed"
+    }
+}
+
+$_scriptsToInstall = New-Object Collections.Generic.List[string]
+
+# Check if all dependency installation scripts have been executed before for this project on this machine
+foreach ($script in $_deps.installDepsScripts) {
+    # For scripts, check if their name is present at the .conf/.depok file to see if they have been executed before
+    $scriptInstalled = (Test-Path ./.conf/.depok) -and (Get-Content ./.conf/.depok | Select-String $script -Quiet)
+
+    if ($scriptInstalled -eq $false) {
+        $_scriptsToInstall.Add($script)
+        Write-Host -ForegroundColor DarkRed "😵 $script dependency installation script not executed before for this project on this machine"
     } else {
-        Write-Host -ForegroundColor DarkGreen "👍 $package installed"
+        Write-Host -ForegroundColor DarkGreen "👍 $script dependency installation script executed before for this project on this machine"
     }
 }
 
-# ask if the user want to install the packages that are not installed
-if ($_packagesToInstall.Count -gt 0) {
+# Check if there are any packages to be installed or scripts to be executed
+if ($_packagesToInstall.Count -eq 0 -and $_scriptsToInstall.Count -eq 0) {
+    Write-Host -ForegroundColor DarkGreen "✅ All packages already installed and installation script executed before for this project on this machine"
```

Comment:
I'm not a big fan of these large messages, could you try to keep it short?

Reply:
Sure